gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java

```diff
@@ -90,6 +90,14 @@ public List<LedgerAddressDTO> get() {
     return result;
   }
 
+  private Long getSetlId(String damlPartyId) {
```

Comment:
I suggest converting this to `Map<String, Long> createPartyToSetlId()` and then on L74 (prior to for-loop) `var partyToSetlId = createPartyToSetlId()`. Furthermore, the construction of such a map could take place in `SetlPartySupplier` and expose a method `Optional<Long> getSetlId(String party)`.

Reply:
I exposed a method that simply returns Long (as we are fine to put null into the result DTO). Is that okay?